src/v/storage/readers_cache.cc

```diff
@@ -80,6 +80,19 @@ readers_cache::put(std::unique_ptr<log_reader> reader) {
     return ptr->make_cached_reader(this);
 }
 
+bool readers_cache::is_within_locked_range(
+  model::offset reader_base_offset, model::offset reader_end_offset) const {
+    auto lock_it = std::find_if(
+      _locked_offset_ranges.begin(),
+      _locked_offset_ranges.end(),
+      [reader_base_offset, reader_end_offset](const offset_range& range) {
+          return reader_base_offset <= range.second
```

Comment:
Good find. This condition looks much correct to me. Previous one didn't make any sense to me.